File: Sqrt/Sqrt.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x <= 1:
            return x
        left = 0
        right = x
        while left < right:
            mid = (left + right) / 2
            if mid * mid > x:
                right = mid
            elif mid * mid < x:
                left = mid + 1
            else:
                return mid
        return left - 1

File: Sqrt/Sqrt.cpp
class Solution {
public:
    int mySqrt(int x) {
        if(x <= 1) {
        	return x;
        }
        int left = 0, right = x, mid;
        while(left < right) {
        	int mid = (left + right) / 2;
        	if(mid > x / mid)
        		right = mid;
        	else if(mid < x / mid)
        		left = mid + 1;
        	else
        		return mid;
        }
        return left - 1;
    }
};